src/track/trackrecord.cpp

```diff
@@ -199,4 +206,44 @@ bool TrackRecord::mergeImportedMetadata(
     return modified;
 }
 
+bool TrackRecord::updateStreamInfoFromSource(
+        const mixxx::audio::StreamInfo& streamInfoFromSource) {
+    // Stream properties are not expected to vary during a session
+    VERIFY_OR_DEBUG_ASSERT(!m_streamInfoFromSource ||
+            *m_streamInfoFromSource == streamInfoFromSource) {
+        kLogger.warning()
+                << "Varying stream properties:"
+                << *m_streamInfoFromSource
+                << "->"
+                << streamInfoFromSource;
+    }
+    m_streamInfoFromSource = streamInfoFromSource;
+    // Stream info from source is always propagated to metadata and
+    // unconditionally overwrites any properties that have once been
+    // parsed from file tags! This is required to store the most
+    // accurate information about the audio stream in the database.
+    const bool metadataUpdated = refMetadata().updateStreamInfoFromSource(streamInfoFromSource);
+    DEBUG_ASSERT(getMetadata().getStreamInfo() == streamInfoFromSource);
+    return metadataUpdated;
+}
+
+bool operator==(const TrackRecord& lhs, const TrackRecord& rhs) {
+    // m_streamInfoFromSource must not be considered for equality
+    // comparisons! It is only needed for verifying consistency
+    // during updates and as a flags when a track is loaded.
```

Comment:
Maybe move it out of the function and use `///` then it might get picked up by doxygen/IDEs.